src/IBM.Cloud.SDK.Core/Authentication/CloudPakForData/CloudPakForDataAuthenticator.cs

```diff
@@ -85,12 +143,45 @@ private void Init(string url, string username, string password, bool? disableSsl
                 Headers = headers;
             }
 
-            Validate();
+            Init();
+        }
+
+        private void Init(string url, string username, string password, string apikey, bool? disableSslVerification = null, Dictionary<string, string> headers = null)

```

Comment:
Do we need default values for password and apikey here? I dont think its guaranteed the values will come back from the config
```suggestion
        private void Init(string url, string username, string password = null, string apikey = null, bool? disableSslVerification = null, Dictionary<string, string> headers = null)

```